sentinel5dl/__init__.py

```diff
@@ -74,13 +84,36 @@ def __http_request(path, filename=None):
     logger.debug(f'Requesting {url}')
     with open(filename, 'wb') if filename else io.BytesIO() as f:
         curl = pycurl.Curl()
+        if ca_info:
+            curl.setopt(pycurl.CAINFO, ca_info)
         curl.setopt(curl.URL, url.encode('ascii', 'ignore'))
 
         curl.setopt(curl.USERPWD, f'{USER}:{PASS}')
         curl.setopt(curl.WRITEDATA, f)
         curl.setopt(curl.FAILONERROR, True)
-        curl.perform()
+
+        # abort if data trnsfer is slower than 30 bytes/sec during 60 seconds
+        curl.setopt(pycurl.LOW_SPEED_TIME, 10)
+        curl.setopt(pycurl.LOW_SPEED_LIMIT, 300)
+
+        # try to execute curl.perform() up to 10 times if it is not working
+        for _ in range(10):
+            try:
+                curl.perform()
+
+            except pycurl.error:
+                # wait before next attempt
+                time.sleep(1)
+            else:
+                # successful http request so dont try again
+                break
+        # else:
+            # maximun amount of http requests failed
+
+            # TODO: how to return failed __http_request??
```

Comment:
Depending on how many times you want to re-try, you could do something like:

```py
try:
    curl.perform()
￼except pycurl.error:
    time.sleep(5)
    curl.perform()
```

which will perform just one retry but is very simple and clean.

The alternative would probably be to use the counter in some way or another:

```py
…
except pycurl.error as err:
    if i >= 9: # on last try
        raise err
    time.sleep(1)
```

I would tend to go with the former one since the requests do not seem to be that fragile and it's less code and overall simpler but both should work and the latter could eben be made configurable in the long run.